src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySink.java

```
@@ -92,6 +107,18 @@ protected BigQuerySinkConfig getConfig() {
   protected void prepareRunValidation(BatchSinkContext context) {
     FailureCollector collector = context.getFailureCollector();
     config.validate(context.getInputSchema(), config.getSchema(collector), collector);
+    try {
+      config.validatePartitionProperties(context.getInputSchema(), config.getSchema(collector), collector);
+    } catch (InvalidConfigPropertyException e) {
+      collector.addFailure(String.format("Unable to load credentials from %s.", config.getServiceAccountFilePath()),
```

is this guaranteed to be a problem loading credentials? I think it's better to just let the exception propagate up if this is at prepare time. Whoever throws it should be using the right message.

the collector is mainly used to collector a bunch of problems for validation when the pipeline is being created, but using it adds a lot of complexity that isn't required if we're already at runtime.

---

In that situation it was, but it is probably worth refactoring the code to remove the complexity. Fixed.